### ISSUE TITLE
Training stuck

### ISSUE THREAD

**Issue by linkailong555-del** [OpenPipe/ART#377]
My rule model is a local model deployed in VLLM, and whenever I need to score after generating trajectories, it gets stuck for a long time:
Training data size: 3683
Training for 3 epoch(s)
Generating 3 responses per input for RULER to compare
============ Training Loop =============
Training: 0it [00:00, ?it/s]
Training step 285:   3%|████▍                                                                                                                                              | 285/9549 [00:00<?, ?batch/s]
gather: 100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 3/3 [00:16<00:00,  5.56s/it, reward=0, completion_tokens=902]

However, no logs were output on the VLLM.

**Comment by h66840** [comment]
There is a high probability that the reason why the training code is stuck is that the call tool in the roll out phase does not create a connection correctly when communicating with the Smithery or the MCP of the local host.
